qhdwight/mrover-workspace
# auton gate search

- Gate Traversal PR Code (#816)
- lkf fixes ryan made (#818)
- Mp9250 imu (#823)
- Update auton-integration to main (#840)
- [nav] Basic Target Caching (#832)
- Added launch capabilities to jarvis (#834)
- GPS Spoof for localization filter (#841)
- Update Auton GUI (#839)
- [nav] Caching System changes (#844)
- Fixed bus in launch script (#845)
- Added default initialization of obstacle lcm (#847)
- Added gdb capabilities to jarvis (#848)
- [nav] Updated nodejs version (#828)
- [bot] Bump py from 1.4.34 to 1.10.0 in /jarvis_files (#664)
- [bot] Bump lodash from 4.17.20 to 4.17.21 in /base_station/gui (#666)
- [bot] Bump chart.js from 2.7.3 to 2.9.4 in /base_station/gui (#672)
- [bot] Bump websockets from 3.4 to 9.1 in /pip_deps (#673)
- [nav] Fixed infinite looping due to caching (#850)
- [bot] Bump tar from 4.4.6 to 4.4.19 in /base_station/gui (#707)
- Flip drive polarity (#857)
- Updated jetson odrive version to 0.5.4 in pip dependencies (#812)
- serial parser for imu (#868)
- [nav] 2 Bearing Obstacle Avoidance (#889)
- Arduino code for sending IMU data (#874)
- fix gdb in jarvis (#873)
- Getting integration up to date with main (#880)
- Added sudo login capabilities (#854)
- Arduino code for new BNO055 IMU (#896)
- Added Nav State Color changing box and made space to add Zed Camera Feed under waypoint Editor (#875)
- [admin] Resolved cython and scipy build issues (#597) (#597)
- Updated jetson odrive_bridge to use firmware version to 0.5.4 (#812)
- [admin] Resolved cython and scipy build issues (#597)
- Jarvis debugger fix (#853)
- [nav] Fixes from rover testing on 2/6 (#901)
- [admin] removed python3.7 from ansible scripts (#921)
- Changed driving power (#925)
- Jarvis launch changes (#932)
- Removal of search spin states, hopeful to reduce time (#934)
- Remove scipy dependency from filter (#930)
- [esw, teleop] Integrate arm controls and IK
- [nav] Updated config  (#939)
- [sim] Percep noise + GPS noise simulator with threshold adjustment (#888)
- [teleop] Auton GUI Updates (#926)
- [nav] Remove Radio Repeater code from nav (#871)
- [loc] Added calibration fields to IMUData struct + IMU now sending magnetometer data (#935)
- [bot] Bump ajv from 6.5.2 to 6.12.6 in /base_station/gui (#909)
- [esw] Rolling rover drive code  (#891)
- [nav] Python Search Point Generation for Spiral-out Search (#938)
- [teleop] added calibration data to auton GUI (#946)
- [GUI] Added zoom beyond native tilemap zoom to auton gui rover map (#948)
- [teleop] Center auton map on rover after first odom lcm message (#949)
- [percep] ar tag fixes (#951)
- [sim] 2D Simulator Field of View Visualization + Enable LCM + Perception Noise Code Reduction (#943)
- 1 bit border (#952)
- [teleop] Add auton reverse toggle (#947)
- changed default mode to pipe
- [loc] changed default mode to pipe (#957)
- [esw] drive code fix for unexpected ACK issue (#956)
- [gui] Add first waypoint lat/lon to logging (#955)
- [admin] Rebased with mrover-workspace/main (#944)
- [nav] Test fixes (#965)
- Added odrive to jarvis launch and adjusted minor bugs (#967)
- Added TargetList to auton data logging (#963)
- [nav] Hit counter for nav (#968)
- fix stuck in turn and drive to target bug (#970)
- [esw] Drive watchdog bug fix (#978)
- [teleop] Auton log reader (#971)
- Remove intermediate, include change for checking if immediately at point (#982)
- Search pattern upgrades (#979)
- [admin] Auton integration rebase (#984)
- [bot] Bump minimist from 1.2.0 to 1.2.6 in /base_station/gui (#976)
- [nav] matt recommended clean up fixes (#945)
- Auton map waypoints now display WaypointName,index (#981)
- [loc] GPS encoder stop, filter crash fix (#988)
- [teleop] Add tangent arrow (#986)
- [teleop] Remove auton reverse toggle and functionality (#993)
- Encoder stop fix (#999)
- flag to get rid of installing pip requirements, jarvis checks the file hash instead of contents....
- [nav] Target in Drive + Other Fixes (#1004)
- [admin] Flag to disable installing pip dependencies  (#1003)
- [teleop] Added intermediate step between toggling auton mode (#998)
- [Auton, Teleop] Auton drive change (#1002)
- [esw] Moved polarity flip to esw as a hardware change since we will never be driving backwards (#1000)
- [teleop, esw] polarity flip (#1013)
- [auton] Navigation New Architecture & Refactoring (#972)
- [teleop] Auton GUI map will now input coordinates to waypoint inputs on click (#1015)
- Added /search_points lcm struct (#1011)
- Changed type of search_pattern_size member of /search_point struct fr… (#1022)
- [admin] Jarvis Launch Additions (#1010)
- [teleop] Receive and show search points in GUI (#1031)
- [nav] Gate Search (#1029)
- [teleop] Drive polarity flips, improved controls (#1012)
- [nav] SearchPoints LCM (#1027)
- small fix (#1034)
- [nav] Projected Points (#1035)
- Fix merge conflicts in teleop main (#1039)
- [teleop] Name Auton report, confirm closing window (#1041)
- [nav, sim] Auton gate search (#1044)
- Don't use std::filesystem for 18.04, set search path to 1 if aligned
- Fixed diamond path, fixed path indexing
- Comment code and refactor
- Add more comments, removed unused things
